### PR TITLE
feat(core): EventEmitter should throw if outside angular zone

### LINKED ISSUE #8927: Throw error when emitting from an @Output when not in dev mode
A common pattern I've seen is an Angular component wrapping a third-party library (e.g. Google Maps or Recaptcha APIs). These libraries are loaded outside of the Angular zone and thus any events that they initiate run outside of it.

Unfortunately, that means that if component attempting to wrap the library naively passes a callback to it, that callback will be executed outside the Angular zone. Frequently such callbacks attempt to emit `@Output` events.

It might be valuable to warn the developer in such cases that this is probably not what they want, by detecting whether the Angular zone is active for an `@Output` `EventEmitter.emit()` call and throwing if it's not. This check could be disabled in production mode, and could also be avoided by constructing an `EventEmitter` with a special flag if emitting events outside the Angular zone is indeed the desired behavior.

What do you think @mhevery @robwormald?

### PR DESCRIPTION
**Pull request by DzmitryShylovich**
Closes #8927

### CODE REVIEW

File: modules/@angular/core/src/zone/ng_zone.ts [hunk 1]
```diff
@@ -6,7 +6,8 @@
  * found in the LICENSE file at https://angular.io/license
  */
 
-import {EventEmitter} from '../event_emitter';
```
Comment (DzmitryShylovich) [inline modules/@angular/core/src/zone/ng_zone.ts]: otherwise it will be a circular dependency

File: modules/@angular/core/src/event_emitter.ts [hunk 2]
```diff
@@ -56,55 +58,51 @@ import {Subject} from 'rxjs/Subject';
...
-  constructor(isAsync: boolean = false) {
-    super();
-    this.__isAsync = isAsync;
-  }
+  constructor(private isAsync: boolean = false) { super(); }
+
+  emit(value?: T): void { this.next(value); }
 
-  emit(value?: T) { super.next(value); }
+  next(value?: T): void {
```
Comment (DzmitryShylovich) [inline modules/@angular/core/src/event_emitter.ts]: some people use `next` instead of `emit`

**Comment by splincode** [comment]
@DzmitryShylovich  what is the status of your pull request? is it still relevant?

**Comment by mhevery** [comment]
closing as old. Also we are thinking how to rely on zone less, and this PR would move it to rely on Zones more.

**Pull request closed without merging**